### DISCUSSION TITLE
CUE documentation should talk about whether terser diffs are selling point

### DISCUSSION THREAD

**Discussion by paul-hammant** [cuelang/cue#832] [Ideas]
You know, like what git-diff would say about an item added to a list - how many lines does the underlying diff engine think has changed

**Comment by xinau** [comment]
Thanks for your suggestion. I think it would be a good idea to add this feature to the documentation once https://github.com/cuelang/cue/issues/8 is implemented :)

**Comment by paul-hammant** [comment]
I don't think so. This is a function of Git, Mercurial, Svn, Perforce, PlasticSCM. How pretty are diffs in that tech (unified diff really), for a number of change scenarios. And I say that becuase the at-rest format for CUE things is carriage-return delimited text (UTF8, or a codepage, etc), is it not??

**Comment by xinau** [comment]
@paul-hammant could you please elaborate a little bit more and provide an example on your issue/request as I'm struggling to understand the underlying issue.

**Comment by paul-hammant** [comment]
$> diff ABCv1.toml ABCv2.toml

What does that diff look like (but CUE source not .txt files)? Is it noisy for a given scenario, or terse?  If XML is OK for diffs (for that scenario). JSON is noisy because of the commas. What does CUE diffs look like?

**Comment by myitcv** [comment]
Thanks for raising this point @paul-hammant - I've converted your original issue to a discussion for better visibility.

**Comment by myitcv** [comment]
To answer your question with a couple of specific points, @paul-hammant. 

[Commas](https://tip.cuelang.org/docs/references/spec/#commas) are required between list elements, but when a list is split across multiple lines `cue fmt` automatically adds a trailing `,` to the last element. For example:

```
x: [
	1,
	2,
	3
]
```

would automatically be converted to:

```
x: [
	1,
	2,
	3,
]
```

[LSP support](https://github.com/cuelang/cue/issues/142) will bring auto-formatting to all LSP-enabled editors. 

CUE also has an internal package for calculating semantic diffs between configurations. We have recently discussed how that might be surfaced as a subcommand of `cmd/cue`. That would be a building block for using such a tool as an alternative `git` diff tool.

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/832.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
